prov/efa/src/rxr/rxr_ep.c

```diff
@@ -2037,6 +2041,12 @@ static inline void rxr_ep_poll_cq(struct rxr_ep *ep,
 		if (OFI_UNLIKELY(ret == 0))
 			return;
 
+		if (is_shm_cq && src_addr != FI_ADDR_UNSPEC) {
+			/* convert SHM address to EFA address */
+			assert(src_addr < EFA_SHM_MAX_AV_COUNT);
+			src_addr = efa_av->shm_rdm_addr_map[src_addr];
```

Comment:
I was going to suggest adding the reverse look up inside `rxr_cq_handle_pkt_recv_completion` along with the `FI_ADDR_NOTAVAIL` handling, but I see you can't tell betwen shm or efa completion by then.

Reply:
You are right, that information is only available in rxr_ep_poll_cq() or in rxr_ep_progress_internal()